### ISSUE TITLE
Disqus Integration for IJP Open articles

### ISSUE THREAD

**Issue by rickardnyman** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#582]
Disqus integration below IJP Open articles with a link to take you down to discussion somewhere at the top of the page. As with the old IJP Open

**Comment by rickardnyman** [comment]
@adistasio @abaker-gavant 

When you get a chance please take a look at feasibility/timeline of disqus feature above?

**Comment by nrshapiro** [comment]
Since I am planning to push to production the server and client on January 22, and we need testing time on Stage, we'd like this done by Jan 7.

**Comment by bakerac4** [comment]
@nrshapiro do you have the account info for disqus?

**Comment by nrshapiro** [comment]
From @bakerac4 (Moving here for discussion to include all parties, e.g., @rickardnyman --who is in charge of the IJPOpen extensions to PEP-Web)  I will add my own thoughts in responses below Adam's message. 

Attached is a mock up of what we are thinking it will look like.

There are 2 reasons we decided to put it in a panel instead of just putting it at the bottom.
-     Due to the scrollbar library we use its technically quite hard as they don’t support iframes, so we would either have to rip it out which would cause churn or make the comments scroll separately (which is a weird UX)
-     This allows the user to comment on an article without having to scroll down all the way which is a better UX

We just wanted to confirm with you guys before we went down this road, so please send me an email with your thoughts.
Thanks,
Adam

<img width="1436" alt="image001" src="https://user-images.githubusercontent.com/4837554/148975089-59d2c9f7-a679-4cf9-88be-fc5223bc5201.png">

**Comment by nrshapiro** [comment]
@bakerac4 @rickardnyman 

I think that's fine as long as:
- the panes are drag/resizeable
- The comment button at top should toggle the pane at the bottom 
- There should be a toggle to maximize (almost full height) and minimize the bottom pane quickly so they can read and type comments more easily, and quickly get back to the article in question.
For reference, this was the view of the articles with commentary in GVPi's interface.  This had the advantage (and disadvantage) that the articles were not part of the interface (since they were in PDF).  That allowed more screen space for viewing comments and composing.  That had the disadvantage though of having the article itself in a different window or app (PDF Viewer).

<img width="322" alt="image" src="https://user-images.githubusercontent.com/4837554/148976140-5aa059d3-f96d-4abf-bcfe-e65c8116923e.png">

**Comment by nrshapiro** [comment]
1) It does not appear to be working.  See below.
2) Also, can we set it to open by default for IJPOpen?  It's easy to miss otherwise.

<img width="938" alt="image" src="https://user-images.githubusercontent.com/4837554/149246423-12ad882e-b351-4f58-8db0-64f2d86aac9e.png">

**Comment by nrshapiro** [comment]
@rickardnyman @bakerac4 

The Disqus integration is working...I don't know what they did except to "repush" the same build.  But now it works.

Overall, it works--the old reviews are visible.  However, the new pane and the resizing controls behave irrationally.

- If I size it, I expected it to come back that size next time.  Clicking on the discussion icons shows it almost always covers the window...though sometimes it doesn't on some documents (and it's not related to the first discussion size, my tests have proven).  I had thought it was trying to base the size on the content, but if you try a few documents, you'll see it doesn't.   In any case, since there's a maximize button, there's no need for such "intelligence".
- If I use the "maximize button on the pane, it grows to almost fill the center pane.  That's fine.  But when I click it again, it doesn't shrink, it grows to almost completely fill the window.  It should toggle between almost maximum, and the size I set.

See the video for a demonstration with one document.

https://user-images.githubusercontent.com/4837554/149419167-7cb227a2-26b8-4c98-91b3-6335d1fcc604.mp4

**Comment by nrshapiro** [comment]
@adistasio @bakerac4 @billdami @rickardnyman @davidtuckett 

Rickard and I went over the Disqus window pane controls and we see it as problematic.

Currently: 

1) When you open an IJPOpen article, the Disqus pane is ALWAYS off by default
2) When you open it clicking the cloud.  It ALWAYS jumps to the larger "fit to size" (FTS) size.  That seems to be a default.
3) When you turn off FTS, the pane gets larger, going over the border on a moderate size screen  You can't drag the window pane anymore since the top is obscured.
4) When you go to another IJPOpen document, the Disqus pane is off by default.
5) If you open it, you are back to FTS, even though you turned that off

Thus it never remembers where you might have resized the window pane.

We think the user settings should remain "constant", putting the user in control over the window size, and whether it's FTS or the maximize/minimize (MM) button is clicked.  These should be sticky.

Here's how we think it should work, for discussion:

1) The IJP Discussion (Thought Cloud) button should be sticky.  If you turn it on, it should stay on until you turn it off
   a) That means you can walk up and down the list of IJPOpen articles on the left, and preview the discussion
   b) Where I set the discussion pane size, should be remembered

2) The FTS button when off should stay off until I turn it back on.  I'm less sure it should stay on when I turn it on, but for consistency, let's say yes.

3. When the FTS button is turned off, the pane size should return to where the user set it, or the default if they did not set it.  In other words, it should be equivalent to "Restore".  (Note the MM button has a Maximize and a Minimize, but not a restore.  That makes this second button important, especially in its ability to Restore (when it works that way.)

4) The app should remember these settings at least for the session...perhaps as part of one of the cookies it stores.  

In short: the goal is to give the user control of the windows and panes.  The buttons will work as above to provide Max, Min, FTS, and Restore functionality with the user in control.

**Comment by nrshapiro** [comment]
To add:

5. Given that users may not notice the cloud icon (I didn't at first), causing issues for support, it may be a good idea to INITIALLY default the discussion button to on, until the user turns it off, and then remember their setting.

**Comment by nrshapiro** [comment]
@abaker-gavant @adistasio @rickardnyman @ocappello 

Important [Confirmed with @davidtuckett]  

6. Only users who are **logged in** should be able to see/open the Disqus panel.

Users needing Disqus based accounts for discussion is ok.  Only PEP logged in users will be able to see the panel so that provides a bit of protection.

**Comment by nrshapiro** [comment]
@abaker-gavant @adistasio @rickardnyman @ocappello @davidtuckett 

If it helps as we move forward, beyond Monday, although the old site is not available, I asked Michael (GVPi) to see if he remembered how it worked.
```
Neil,

1. The site sent an automatic login to Disqus so that the user didn’t have to login again using the method at https://help.disqus.com/en/articles/1717160-integrating-single-sign-on.

2. I don’t have a memory of when exactly Disqus was displayed.  Looking at the code, the old site showed the Disqus section on IJPOpen pages when there was a user logged in.  I can’t quite identify if there was a requirement for an IJPOpen access record as well.

Michael```
